R/template.R

```diff
@@ -266,133 +265,161 @@ prob_estimate_convert <- function(estimate) {
+#'   method of your metric function.
 #'
-#' @param truth The realized vector of `truth`. This is either a factor
-#' or a numeric.
+#' @param truth The realized vector of `truth`. This is either a factor or a
+#'   numeric.
 #'
 #' @param estimate The realized `estimate` result. This is either a numeric
-#' vector, a factor vector, or a numeric matrix (in the case of multiple
-#' class probability columns) depending on your metric function.
+#'   vector, a factor vector, or a numeric matrix (in the case of multiple class
+#'   probability columns) depending on your metric function.
 #'
 #' @param na_rm A `logical` value indicating whether `NA` values should be
-#' stripped before the computation proceeds. `NA` values are removed
-#' before getting to your core implementation function so you do not have to
-#' worry about handling them yourself. If `na_rm=FALSE` and any `NA` values
-#' exist, then `NA` is automatically returned.
+#'   stripped before the computation proceeds. `NA` values are removed before
+#'   getting to your core implementation function so you do not have to worry
+#'   about handling them yourself. If `na_rm=FALSE` and any `NA` values exist,
+#'   then `NA` is automatically returned.
 #'
-#' @param cls A character vector of length 1 or 2 corresponding to the
-#' class that `truth` and `estimate` should be, respectively. If `truth` and
-#' `estimate` are of the same class, just supply a vector of length 1. If
-#' they are different, supply a vector of length 2. For matrices, it is best
-#' to supply `"numeric"` as the class to check here.
-#'
-#' @param estimator The type of averaging to use. By this point, the averaging
-#' type should be finalized, so this should be a character vector of length 1\.
-#' By default, this character value is required to be one of: `"binary"`,
-#' `"macro"`, `"micro"`, or `"macro_weighted"`. If your metric allows more
-#' or less averaging methods, override this with `averaging_override`.
-#'
-#' @param case_weights Optionally, the realized case weights, as a numeric
-#' vector. This must be the same length as `truth`, and will be considered in
-#' the `na_rm` checks. If supplied, this will be passed on to `metric_impl` as
-#' the named argument `case_weights`.
+#' @param case_weights the realized case weights, as a numeric vector. This must
+#'   be the same length as `truth`, and will be considered in the `na_rm`
+#'   checks. If supplied, this will be passed on to `metric_impl` as the named
+#'   argument `case_weights`.
 #'
 #' @param ... Extra arguments to your core metric function, `metric_impl`, can
-#' technically be passed here, but generally the extra args are added through
-#' R's scoping rules because the core metric function is created on the fly
-#' when the vector method is called.
+#'   technically be passed here, but generally the extra args are added through
+#'   R's scoping rules because the core metric function is created on the fly
+#'   when the vector method is called.
```

Comment:
This is the craziness you'd be able to eliminate by letting the metric functions call their "core metric functions" themselves. Wouldn't need any reliance on scoping rules, and would be way more R like